### PR TITLE
Refactor/merge contexts

### PR DESCRIPTION
**Pull request by FilipMessa**
- Merge BookingContext into the SearchContext
- Prerender PassengerCards

### CODE REVIEW

**Comment by FilipMessa** [comment]
PLEASE don't merge if it hasn't at least two approvals

File: apps/core/contexts/searchContext/helpers.js [hunk 1]
```diff
@@ -26,7 +27,7 @@ const PARSER_CONFIG = {
   isNightsInDestinationSelected: 'boolean',
 };
 
-export function parseURLqueryToState(query: Object) {
+export function parseURLqueryToState(query: any) {
```
Comment (JoseDuSV) [inline apps/core/contexts/searchContext/helpers.js]: Is that change from `Object` to `any` necessary? Not that `Object` is that much better. But if we cannot use some more exact type it's probably ok.

Comment (FilipMessa) [reply apps/core/contexts/searchContext/helpers.js]: Good point, I try to handle the flow and forgotten to change it back.

File: apps/core/components/passengerForm/PassengerForm.js [hunk 2]
```diff
@@ -289,7 +290,7 @@ class PassengerForm extends React.Component<Props, State> {
             />
             <TextInput
               onChangeText={this.handleNameChange}
-              label="Given names"
+              label="First names"
```
Comment (jvaclavik) [inline apps/core/components/passengerForm/PassengerForm.js]: It seems to me that it should be "first name", without "s", what do you think?

Comment (jvaclavik) [reply apps/core/components/passengerForm/PassengerForm.js]: or maybe first name(s)

Comment (FilipMessa) [reply apps/core/components/passengerForm/PassengerForm.js]: 👍

File: apps/core/contexts/searchContext/helpers.js (hunk 1 above)
Comment (FilipMessa) [reply apps/core/contexts/searchContext/helpers.js]: done

File: apps/core/components/passengerForm/PassengerForm.js (hunk 2 above)
Comment (FilipMessa) [reply apps/core/components/passengerForm/PassengerForm.js]: done

File: apps/core/contexts/searchContext/helpers.js [hunk 3]
```diff
@@ -93,3 +95,48 @@ function booleanParser(boolean: boolean) {
   // @TODO input validation
   return boolean;
 }
+
+function createPassneger(type: 'adult' | 'infant'): PassengerType {
```
Comment (jvaclavik) [inline apps/core/contexts/searchContext/helpers.js]: `createPassneger` :D are you a racist?

Comment (FilipMessa) [reply apps/core/contexts/searchContext/helpers.js]: :D 🙃

**Pull request merged**